### DISCUSSION TITLE
I need to set VGA resolution for 16:9 screen? Can you tell me the maximum resolution I can set?

### DISCUSSION THREAD

**Discussion by 22kashaf-khan** [fdivitto/FabGL#256] [Q&A] [answered]
For now I have seen 640x480, I would like to set it to1280x720, if possible.

**Comment by fdivitto** [comment]
How many colors do you need?

BTW: please post questions in "Discussions"...

**Reply by 22kashaf-khan** [reply]
Maximum 64 colors.

**Comment by fdivitto** [comment, marked as answer]
1280x720x6 (6 bit required to have 64 colors) resolution needs 691200 bytes of internal RAM, too much for the ESP32...

**Reply by 22kashaf-khan** [reply]
SO can you suggest a better alternative resolution or colors?

**Reply by fdivitto** [reply]
It depends by your application memory requirements. I guess you shouldn't get over 640x400 (or 480) with 16 colors (VGA16Controller), as many fabgl demos do.

**Reply by 22kashaf-khan** [reply]
Okay, thank you.

**Reply by 22kashaf-khan** [reply]
Using a model with PSRAM can I achieve a resolution of 1280x720 with 26 colors?

**Reply by fdivitto** [reply]
> Using a model with PSRAM can I achieve a resolution of 1280x720 with 26 colors?

unfortunately no. The frame buffer must be allocated in internal memory.

**Reply by 22kashaf-khan** [reply]
Can I change any hardcoded values in your library to get 1280x720 instead of 640x480?

**Reply by fdivitto** [reply]
resolutions aren't hard coded. FabGL uses modelines (like linux modelines) to define resolution and VGA timings. Please look at fabglconf.h: here there a some modelines, but you can write your own. 
You may test your modelines using "modelines" example.

**Reply by 22kashaf-khan** [reply]
Thank you so much.
